src/ripple/beast/core/CurrentThreadName.cpp

```diff
@@ -0,0 +1,149 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2013, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Portions of this file are from JUCE.
+    Copyright (c) 2013 - Raw Material Software Ltd.
+    Please visit http://www.juce.com
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/beast/core/CurrentThreadName.h>
+#include <boost/thread/tss.hpp>
+
+namespace beast {
+namespace detail {
+
+static boost::thread_specific_ptr<std::string> threadName;
+
+void saveThreadName (std::string name)
+{
+    try
+    {
+        threadName.reset (new std::string {std::move(name)});
+    }
+    catch(...)
+    {
+    }
```

Comment:
I don't really see a strong motivation to swallow exceptions here.  If an exception is thrown here it is likely because we're running out of memory, in which case we're already hosed.  However I don't feel strongly that this behavior needs to be changed here and now.  Might be wise to change that behavior in a separate P/R should we choose to do so.  I'm fine with a comment, not sure what it would say...  `// Questionable try/catch` ?